pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1546,29 +1471,21 @@ func (l *LibvirtDomainManager) SignalShutdownVMI(vmi *v1.VirtualMachineInstance)
 	}
 
 	if domState == libvirt.DOMAIN_RUNNING || domState == libvirt.DOMAIN_PAUSED {
-		domSpec, err := l.getDomainSpec(dom)
-		if err != nil {
-			log.Log.Object(vmi).Reason(err).Error("Unable to retrieve domain xml")
-			return err
-		}
-
 		err = dom.ShutdownFlags(libvirt.DOMAIN_SHUTDOWN_ACPI_POWER_BTN)
 		if err != nil {
 			log.Log.Object(vmi).Reason(err).Error("Signalling graceful shutdown failed.")
 			return err
 		}
 		log.Log.Object(vmi).Infof("Signaled graceful shutdown for %s", vmi.GetObjectMeta().GetName())
 
-		if domSpec.Metadata.KubeVirt.GracePeriod.DeletionTimestamp == nil {
-			now := metav1.Now()
-			domSpec.Metadata.KubeVirt.GracePeriod.DeletionTimestamp = &now
-			d, err := l.setDomainSpecWithHooks(vmi, domSpec)
-			if err != nil {
-				log.Log.Object(vmi).Reason(err).Error("Unable to update grace period start time on domain xml")
-				return err
+		l.metadataCache.GracePeriod.WithSafeBlock(func(gracePeriodMetadata *api.GracePeriodMetadata, _ bool) {
+			if gracePeriodMetadata.DeletionTimestamp == nil {
+				now := metav1.Now()
+				gracePeriodMetadata.DeletionTimestamp = &now
+
+				l.metadataCache.Notify()
```

Comment:
I understand that the notify is non-blocking. I think i'd rather see it outside of the mutex held by `WithSafeBlock` though.